### PR TITLE
Add example config for flink 1.2

### PR DESCRIPTION
**Pull request by prymitive**
Metrics exposed in flink 1.2 are different than those from earlier versions (docs: https://ci.apache.org/projects/flink/flink-docs-release-1.2/monitoring/metrics.html#system-metrics), add a new config covering both task and jobmanager

@brian-brazil

### CODE REVIEW

**Comment by brian-brazil** [comment]
This can be collapsed down further, see the existing flink conifg as an example.

The naming of the files should also be changed to follow what we did for kafka.

**Comment by prymitive** [comment]
Updated with most recent version I'm using, hope it's better

File: example_configs/flink-1-2.yml [hunk 1]
```diff
@@ -0,0 +1,121 @@
+rules:
+  ## Job Manager ##
+  - pattern: org\.apache\.flink\.jobmanager\.([a-zA-Z\-\_]+)\.([a-zA-Z\-\_]+)\.([a-zA-Z\-\_]+)\.([a-zA-Z\-\_]+)\.([a-zA-Z\-\_]+)<host=(.*)><>(Count|Value|Rate)
```
Comment (brian-brazil) [inline example_configs/flink-1-2.yml]: You only want the Count and Value.

Rate can be better calculated on the Prometheus end from the Count.

File: example_configs/flink-1-2.yml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    name: flink_jobmanager_$1_$2_$3_$4_$5
```
Comment (brian-brazil) [inline example_configs/flink-1-2.yml]: The jmx exporter will put in underscores for invalid characters, you can go just one capture group for $1-$5

File: example_configs/flink-1-2.yml [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    type: GAUGE
+    labels:
+      host: $6
```
Comment (brian-brazil) [inline example_configs/flink-1-2.yml]: What is in this host label?

File: example_configs/flink-1-2.yml [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+  - pattern: org\.apache\.flink\.taskmanager\.job\.task\.operator\.(KafkaConsumer|KafkaProducer)\.([a-zA-Z\-\_]+).([a-zA-Z\-\_]+)-([0-9]+)<task_id=(.*), subtask_index=(.*), job_id=(.*), tm_id=(.*), task_attempt_id=(.*), job_name=(.*), operator_name=(.*), task_attempt_num=(.*), host=(.*), task_name=(.*)><>(Count|Value|Rate)
+    name: flink_taskmanager_job_task_operator_$1_$2
+    type: GAUGE
+    labels:
+      topic: $3
+      partition: $4
+      subtask_index: $6
+      job_id: $7
+      tm_id: $8
+      job_name: $10
```
Comment (brian-brazil) [inline example_configs/flink-1-2.yml]: Do the job_id and job_name provide the same information?

**Pull request closed without merging**